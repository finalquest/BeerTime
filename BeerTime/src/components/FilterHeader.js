import React from 'react';
import {
  View,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';
import PropertyLabel from './PropertyLabel';


const styles = StyleSheet.create({
  container: {
    height: 50,
    alignSelf: 'stretch',
    backgroundColor: 'white',
  },
});


const FilterHeader = ({ amount }) => (
  <View style={styles.container}>
    <PropertyLabel
      name="Total beers"
      style={{ flex: 1, alignItems: 'center' }}
      nameStyles={{ fontWeight: '600' }}
      value={amount.toString()}
      separator=":"
    />
  </View>
);
FilterHeader.propTypes = {
  amount: PropTypes.number,
};
FilterHeader.defaultProps = {
  amount: 0,
};

export default FilterHeader;
