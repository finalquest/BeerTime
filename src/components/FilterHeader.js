import React from 'react';
import {
  View,
  StyleSheet,
  Image,
  TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';
import PropertyLabel from './PropertyLabel';

import Box from '../../res/emptyBox.png';
import Empty from '../../res/starEmpty.png';
import Star from '../../res/star.png';


const styles = StyleSheet.create({
  container: {
    height: 50,
    alignSelf: 'stretch',
    backgroundColor: 'white',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  icon: {
    alignSelf: 'center',
    margin: 10,
    width: 25,
    height: 25,
    resizeMode: 'contain',
  },
});


const FilterHeader = ({
  amount, onFilterPressed, onFavPressed, showFav,
}) => {
  const Source = showFav ? Star : Empty;
  return (
    <View style={styles.container}>
      <PropertyLabel
        name="Total beers"
        style={{ flex: 1, alignItems: 'center' }}
        nameStyles={{ fontWeight: '600', marginLeft: 10 }}
        value={amount.toString()}
        separator=":"
      />
      <View style={{ flexDirection: 'row' }}>
        <TouchableOpacity onPress={onFavPressed}>
          <Image source={Source} style={styles.icon} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onFilterPressed}>
          <Image source={Box} style={styles.icon} />
        </TouchableOpacity>
      </View>

    </View>
  );
};
FilterHeader.propTypes = {
  amount: PropTypes.number,
  onFilterPressed: PropTypes.func.isRequired,
  onFavPressed: PropTypes.func.isRequired,
  showFav: PropTypes.bool.isRequired,
};
FilterHeader.defaultProps = {
  amount: 0,
};

export default FilterHeader;
