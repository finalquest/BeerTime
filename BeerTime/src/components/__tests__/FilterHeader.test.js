import React from 'react';
import { shallow } from 'enzyme';

import FilterHeader from '../FilterHeader';

describe('FilterHeader test suite', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<FilterHeader amount={2} />);
    expect(wrapper).toMatchSnapshot();
  });
});
