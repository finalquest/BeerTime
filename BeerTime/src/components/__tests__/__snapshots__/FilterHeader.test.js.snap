// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterHeader test suite renders as expected 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "white",
      "height": 50,
    }
  }
>
  <PropertyLabel
    Component={[Function]}
    ellipsizeMode="tail"
    fontSize={14}
    name="Total beers"
    nameStyles={
      Object {
        "fontWeight": "600",
      }
    }
    numberOfLines={1}
    separator=":"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
    value="2"
    valueStyles={
      Object {
        "flexShrink": 1,
        "textAlign": "center",
      }
    }
  />
</View>
`;
